src/test/java/picard/TestDataProviders.java

```diff
@@ -21,21 +26,46 @@
 public class TestDataProviders {
 
     @Test
-    public void testAllDataProviders() throws IllegalAccessException, InstantiationException, InvocationTargetException {
-        int i = 0;
+    public void IndependentTestOfDataProviderTest() throws IllegalAccessException, InvocationTargetException, InstantiationException {
+        testAllDataProvidersdata();
+    }
+
+    @DataProvider(name = "DataprovidersThatDontTestThemselves")
+    public Iterator<Object[]> testAllDataProvidersdata() throws IllegalAccessException, InstantiationException, InvocationTargetException {
+
+        List<Object[]> data = new ArrayList<>();
         final ClassFinder classFinder = new ClassFinder();
         classFinder.find("picard", Object.class);
-        for (Class testClass : classFinder.getClasses()) {
+
+        for (final Class<?> testClass : classFinder.getClasses()) {
             if (Modifier.isAbstract(testClass.getModifiers())) continue;
             for (final Method method : testClass.getMethods()) {
                 if (method.isAnnotationPresent(DataProvider.class)) {
-                    System.err.println("Method: " + testClass.getName() + ":" + method.getName());
-                    method.invoke(testClass.newInstance());
-                    i++;
+                    data.add(new Object[]{method, testClass});
                 }
             }
         }
-        Assert.assertNotSame(i,0);
-        System.err.println("Found: "+ i + " @DataProviders.");
+        Assert.assertTrue(data.size() > 1);
+        Assert.assertEquals(data.stream().filter(c -> ((Method) c[0]).getName().equals("testAllDataProvidersdata")).count(), 1);
+
+        return data.iterator();
+    }
+
+    // @NoInjection annotations required according to this test:
+    // https://github.com/cbeust/testng/blob/master/src/test/java/test/inject/NoInjectionTest.java
+    @Test(dataProvider = "DataprovidersThatDontTestThemselves")
```

Comment:
Why write this using a data provider? Was there an issue with this being written as a test without a data provider?

Reply:
more meta...

I wanted to be sure that the test itself is working...so this is how it tests itself.

Reply:
Personally I would wait until the test fails in some way, and then add a check for that failure (such as if you weren't getting any dataproviders, you'd add an assert like the one you added on line 48).

You could spend a lot of time testing your test code but IMO it's not where I'd spend my time, especially when it comes to testing for failures that have not occurred yet.

Reply:
not sure what you are referring to....but untested code has bugs, by definition. since the small testing framework I built for testing the DataProviders is code...I wanted to be sure that it works...thus a test. In similar additions in htsjdk I found many DataProviders that were silently failing....so I don't think that this is pre-mature.